from PyQt5 import QtWidgets,uic,QtCore,QtGui
from PyQt5.QtWidgets import QMainWindow

def back():
    ui.lineEdit.setReadOnly(False)
    a=ui.lineEdit.text()
    a=a[0:-1]
    ui.lineEdit.setText(a)
    ui.lineEdit.setReadOnly(True)

def oagain():
    ui.lineEdit.setReadOnly(False)
    b=ui.lineEdit.text()
    b=""
    ui.lineEdit.setText(b)
    ui.lineEdit.setReadOnly(True)
    ui.Button_plus.setEnabled(False)
    ui.Button_mod.setEnabled(False)
    ui.Button_mul.setEnabled(False)
    ui.Button_divide.setEnabled(False)
    ui.Button_7.setEnabled(True)
    ui.Button_8.setEnabled(True)
    ui.Button_9.setEnabled(True)
    ui.Button_4.setEnabled(True)
    ui.Button_5.setEnabled(True)
    ui.Button_6.setEnabled(True)
    ui.Button_1.setEnabled(True)
    ui.Button_2.setEnabled(True)
    ui.Button_3.setEnabled(True)
    ui.Button_0.setEnabled(True)

def seven():
    ui.lineEdit.setReadOnly(False)
    ui.lineEdit.insert("7" )
    ui.lineEdit.setReadOnly(True)
    ui.Button_plus.setEnabled(True)
    ui.Button_mod.setEnabled(True)
    ui.Button_mul.setEnabled(True)
    ui.Button_divide.setEnabled(True)
def eight():
    ui.lineEdit.setReadOnly(True)
    ui.Button_plus.setEnabled(True)
    ui.Button_mod.setEnabled(True)
    ui.Button_mul.setEnabled(True)
    ui.Button_divide.setEnabled(True)
    ui.lineEdit.setReadOnly(False)
    ui.lineEdit.insert("8" )
    ui.lineEdit.setReadOnly(True)
def nine():
    ui.lineEdit.setReadOnly(True)
    ui.Button_plus.setEnabled(True)
    ui.Button_mod.setEnabled(True)
    ui.Button_mul.setEnabled(True)
    ui.Button_divide.setEnabled(True)
    ui.lineEdit.setReadOnly(False)
    ui.lineEdit.insert("9" )
    ui.lineEdit.setReadOnly(True)
def plus():
    ui.Button_7.setEnabled(True)
    ui.Button_8.setEnabled(True)
    ui.Button_9.setEnabled(True)
    ui.Button_4.setEnabled(True)
    ui.Button_5.setEnabled(True)
    ui.Button_6.setEnabled(True)
    ui.Button_1.setEnabled(True)
    ui.Button_2.setEnabled(True)
    ui.Button_3.setEnabled(True)
    ui.Button_0.setEnabled(True)
    ui.lineEdit.setReadOnly(True)
    ui.lineEdit.setReadOnly(False)
    f()
    ui.lineEdit.insert("+" )
    ui.lineEdit.setReadOnly(True)
def four():
    ui.lineEdit.setReadOnly(True)
    ui.Button_plus.setEnabled(True)
    ui.Button_mod.setEnabled(True)
    ui.Button_mul.setEnabled(True)
    ui.Button_divide.setEnabled(True)
    ui.lineEdit.setReadOnly(False)
    ui.lineEdit.insert("4" )
    ui.lineEdit.setReadOnly(True)
def five():
    ui.lineEdit.setReadOnly(True)
    ui.Button_plus.setEnabled(True)
    ui.Button_mod.setEnabled(True)
    ui.Button_mul.setEnabled(True)
    ui.Button_divide.setEnabled(True)
    ui.lineEdit.setReadOnly(False)
    ui.lineEdit.insert("5" )
    ui.lineEdit.setReadOnly(True)
def six():
    ui.lineEdit.setReadOnly(True)
    ui.Button_plus.setEnabled(True)
    ui.Button_mod.setEnabled(True)
    ui.Button_mul.setEnabled(True)
    ui.Button_divide.setEnabled(True)
    ui.lineEdit.setReadOnly(False)
    ui.lineEdit.insert("6" )
    ui.lineEdit.setReadOnly(True)
def mod():
    ui.Button_7.setEnabled(True)
    ui.Button_8.setEnabled(True)
    ui.Button_9.setEnabled(True)
    ui.Button_4.setEnabled(True)
    ui.Button_5.setEnabled(True)
    ui.Button_6.setEnabled(True)
    ui.Button_1.setEnabled(True)
    ui.Button_2.setEnabled(True)
    ui.Button_3.setEnabled(True)
    ui.Button_0.setEnabled(True)
    ui.lineEdit.setReadOnly(True)
    ui.lineEdit.setReadOnly(False)
    f()
    ui.lineEdit.insert("%" )
    ui.lineEdit.setReadOnly(True)
def one():
    ui.lineEdit.setReadOnly(True)
    ui.Button_plus.setEnabled(True)
    ui.Button_mod.setEnabled(True)
    ui.Button_mul.setEnabled(True)
    ui.Button_divide.setEnabled(True)
    ui.lineEdit.setReadOnly(False)
    ui.lineEdit.insert("1" )
    ui.lineEdit.setReadOnly(True)
def two():
    ui.lineEdit.setReadOnly(True)
    ui.Button_plus.setEnabled(True)
    ui.Button_mod.setEnabled(True)
    ui.Button_mul.setEnabled(True)
    ui.Button_divide.setEnabled(True)
    ui.lineEdit.setReadOnly(False)
    ui.lineEdit.insert("2" )
    ui.lineEdit.setReadOnly(True)
    
def thre():
    ui.lineEdit.setReadOnly(True)
    ui.Button_plus.setEnabled(True)
    ui.Button_mod.setEnabled(True)
    ui.Button_mul.setEnabled(True)
    ui.Button_divide.setEnabled(True)
    ui.lineEdit.setReadOnly(False)
    ui.lineEdit.insert("3" )
    ui.lineEdit.setReadOnly(True)
def mul():
    ui.Button_7.setEnabled(True)
    ui.Button_8.setEnabled(True)
    ui.Button_9.setEnabled(True)
    ui.Button_4.setEnabled(True)
    ui.Button_5.setEnabled(True)
    ui.Button_6.setEnabled(True)
    ui.Button_1.setEnabled(True)
    ui.Button_2.setEnabled(True)
    ui.Button_3.setEnabled(True)
    ui.Button_0.setEnabled(True)
    ui.lineEdit.setReadOnly(False)
    f()
    ui.lineEdit.insert("*" )
    ui.lineEdit.setReadOnly(True)
def dec():
    ui.lineEdit.setReadOnly(False)
    ui.lineEdit.insert("." )
    ui.lineEdit.setReadOnly(True)
def zero():
    
    ui.Button_plus.setEnabled(True)
    ui.Button_mod.setEnabled(True)
    ui.Button_mul.setEnabled(True)
    ui.Button_divide.setEnabled(True)
    ui.lineEdit.setReadOnly(False)
    ui.lineEdit.insert("0" )
    ui.lineEdit.setReadOnly(True)
def divide():
    ui.Button_7.setEnabled(True)
    ui.Button_8.setEnabled(True)
    ui.Button_9.setEnabled(True)
    ui.Button_4.setEnabled(True)
    ui.Button_5.setEnabled(True)
    ui.Button_6.setEnabled(True)
    ui.Button_1.setEnabled(True)
    ui.Button_2.setEnabled(True)
    ui.Button_3.setEnabled(True)
    ui.Button_0.setEnabled(True)
    ui.lineEdit.setReadOnly(False)
    f()
    ui.lineEdit.insert("/" )
    ui.lineEdit.setReadOnly(True)
def enter():
    global l,num_1
    ui.lineEdit.setReadOnly(False)
    a=ui.lineEdit.text()
    print(num1)
    print(l)
    print(a)
    print(a[0])
    num2=float(a[(l+1):])
    print(num2)
    if a[l]=='+':
        k=num1+num2
    elif a[l]=='*':
        k=num1*num2
    elif a[l]=='/':
        k=num1/num2
    elif a[l]=='%':
        k=num1%num2
    ui.lineEdit.setText(str(k))
    ui.lineEdit.setReadOnly(True)
    ui.Button_7.setEnabled(False)
    ui.Button_8.setEnabled(False)
    ui.Button_9.setEnabled(False)
    ui.Button_4.setEnabled(False)
    ui.Button_5.setEnabled(False)
    ui.Button_6.setEnabled(False)
    ui.Button_1.setEnabled(False)
    ui.Button_2.setEnabled(False)
    ui.Button_3.setEnabled(False)
    ui.Button_0.setEnabled(False)

def f():
    global num1,l
    num1=ui.lineEdit.text()
    l=len(num1)
    num1=float(num1)
    print(num1)
    
num1=00
l=len(str(num1))
app=QtWidgets.QApplication([])
ui=uic.loadUi("Calc.ui")
ui.setWindowIcon(QtGui.QIcon("download.png"))
ui.setWindowTitle("Our calculator")
ui.Button_plus.setEnabled(False)
ui.Button_mod.setEnabled(False)
ui.Button_mul.setEnabled(False)
ui.Button_divide.setEnabled(False)
ui.Button_back.setShortcut("Backspace")
ui.Button_7.setShortcut("7")
ui.Button_8.setShortcut("8")
ui.Button_9.setShortcut("9")
ui.Button_6.setShortcut("6")
ui.Button_5.setShortcut("5")
ui.Button_4.setShortcut("4")
ui.Button_3.setShortcut("3")
ui.Button_2.setShortcut("2")
ui.Button_1.setShortcut("1")
ui.Button_0.setShortcut("0")
ui.Button_plus.setShortcut("+")
ui.Button_mod.setShortcut("Shift+5")
ui.Button_mul.setShortcut("*")
ui.ButtonEnter.setShortcut("Return")
ui.Button_divide.setShortcut("/")
ui.Button_dot.setShortcut(".")
ui.Button_back.clicked.connect(back)
ui.Button_7.clicked.connect(seven)
ui.Button_8.clicked.connect(eight)
ui.Button_9.clicked.connect(nine)
ui.Button_6.clicked.connect(six)
ui.Button_5.clicked.connect(five)
ui.Button_4.clicked.connect(four)
ui.Button_3.clicked.connect(thre)
ui.Button_2.clicked.connect(two)
ui.Button_1.clicked.connect(one)
ui.Button_0.clicked.connect(zero)
ui.Button_plus.clicked.connect(plus)
ui.Button_mod.clicked.connect(mod)
ui.Button_mul.clicked.connect(mul)
ui.ButtonEnter.clicked.connect(enter)
ui.Button_divide.clicked.connect(divide)
ui.Button_dot.clicked.connect(dec)
ui.pushButton_2.clicked.connect(oagain)
ui.show()
app.exec_()
